onnxruntime/core/framework/tensorprotoutils.cc

```diff
@@ -1034,41 +1113,42 @@ template common::Status GetSizeInBytesFromTensorProto<kAllocAlignment>(const ONN
 template common::Status GetSizeInBytesFromTensorProto<0>(const ONNX_NAMESPACE::TensorProto& tensor_proto, size_t* out);
 
-#define CASE_UNPACK(TYPE, ELEMENT_TYPE, DATA_SIZE)                                         \
-  case ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_##TYPE: {                \
-    if (initializer.data_location() == TensorProto_DataLocation_EXTERNAL) {                \
-      ORT_RETURN_IF_ERROR(ReadExternalDataForTensor(                                       \
-          initializer,                                                                     \
-          model_path.IsEmpty() ? nullptr : model_path.ParentPath().ToPathString().c_str(), \
-          unpacked_tensor,                                                                 \
-          tensor_byte_size));                                                              \
-      tensor_data_length = tensor_byte_size;                                               \
-      return Status::OK();                                                                 \
-    } else {                                                                               \
-      size_t element_count = 0;                                                            \
-      if (initializer.has_raw_data()) {                                                    \
-        tensor_byte_size = initializer.raw_data().size();                                  \
-        element_count = tensor_byte_size / sizeof(ELEMENT_TYPE);                           \
-      } else {                                                                             \
-        element_count = initializer.DATA_SIZE();                                           \
-        tensor_byte_size = element_count * sizeof(ELEMENT_TYPE);                           \
-      }                                                                                    \
-      tensor_data_length = tensor_byte_size;                                               \
-      unpacked_tensor.reset(new uint8_t[tensor_data_length]);                              \
-      return onnxruntime::utils::UnpackTensor(                                             \
-          initializer,                                                                     \
-          initializer.has_raw_data() ? initializer.raw_data().data() : nullptr,            \
-          initializer.has_raw_data() ? initializer.raw_data().size() : 0,                  \
-          reinterpret_cast<ELEMENT_TYPE*>(unpacked_tensor.get()), element_count);          \
-    }                                                                                      \
-    break;                                                                                 \
+#define CASE_UNPACK(TYPE, ELEMENT_TYPE, DATA_SIZE)                              \
+  case ONNX_NAMESPACE::TensorProto_DataType::TensorProto_DataType_##TYPE: {     \
```

Comment:
could use ToTensorProtoElementType<ELEMENT_TYPE>() and remove the TYPE argument

Reply:
We could, but that requires someone to potentially go and find the definition of ToTensorProtoElementType to understand what it returns. Given that so I'll leave it with the simpler existing code for now.